docker.go

```diff
@@ -270,6 +301,35 @@ func (p Plugin) Exec() error {
 	return nil
 }
 
+// helper function to set the credentials
+func setDockerAuth(username, password, registry, baseImageUsername,
+	baseImagePassword, baseImageRegistry string) ([]byte, error) {
+	var credentials []docker.RegistryCredentials
+	// add only docker registry to the config
+	if password != "" && strings.Contains(registry, "docker") {
+		dockerConfig := docker.NewConfig()
```

Comment:
`dockerConfig` must first be first defined outside of the `if` condition. Currently this causes an error
```
./docker.go:310:3: dockerConfig declared and not used
```

Reply:
Yea, the latest commit hasn't been pushed. This is already addressed.